File: cslavb/Csla/Serialization/BinaryFormatterWrapper.vb
Imports System.Runtime.Serialization.Formatters.Binary

Namespace Serialization

  ''' <summary>
  ''' Wraps the <see cref="BinaryFormatter"/>
  ''' in the 
  ''' <see cref="ISerializationFormatter"/>
  ''' interface so it can be used in a standardized
  ''' manner.
  ''' </summary>
  Public Class BinaryFormatterWrapper
    Implements ISerializationFormatter

    Private _formatter As BinaryFormatter = New BinaryFormatter()

#Region "ISerializationFormatter Members"

    ''' <summary>
    ''' Converts a serialization stream into an
    ''' object graph.
    ''' </summary>
    ''' <param name="serializationStream">
    ''' Byte stream containing the serialized data.</param>
    ''' <returns>A deserialized object graph.</returns>
    Public Function Deserialize(ByVal serializationStream As System.IO.Stream) As Object _
      Implements ISerializationFormatter.Deserialize

      Return _formatter.Deserialize(serializationStream)

    End Function

    ''' <summary>
    ''' Converts an object graph into a byte stream.
    ''' </summary>
    ''' <param name="serializationStream">
    ''' Stream that will contain the the serialized data.</param>
    ''' <param name="graph">Object graph to be serialized.</param>
    Public Sub Serialize(ByVal serializationStream As System.IO.Stream, ByVal graph As Object) _
      Implements ISerializationFormatter.Serialize

      _formatter.Serialize(serializationStream, graph)

    End Sub

#End Region

    ''' <summary>
    ''' Gets a reference to the underlying
    ''' <see cref="BinaryFormatter"/>
    ''' object.
    ''' </summary>
    Public ReadOnly Property Formatter() As BinaryFormatter
      Get
        Return _formatter
      End Get
    End Property
  End Class

End Namespace

File: Source/Csla.Shared/Serialization/BinaryFormatterWrapper.cs
#if !NETFX_CORE
//-----------------------------------------------------------------------
// <copyright file="BinaryFormatterWrapper.cs" company="Marimer LLC">
//     Copyright (c) Marimer LLC. All rights reserved.
//     Website: http://www.lhotka.net/cslanet/
// </copyright>
// <summary>Wraps the <see cref="BinaryFormatter"/></summary>
//-----------------------------------------------------------------------
using System;
using System.Runtime.Serialization.Formatters.Binary;

namespace Csla.Serialization
{
  /// <summary>
  /// Wraps the <see cref="BinaryFormatter"/>
  /// in the 
  /// <see cref="ISerializationFormatter"/>
  /// interface so it can be used in a standardized
  /// manner.
  /// </summary>
  public class BinaryFormatterWrapper : ISerializationFormatter
  {
    private BinaryFormatter _formatter =
      new BinaryFormatter();

    #region ISerializationFormatter Members

    /// <summary>
    /// Converts a serialization stream into an
    /// object graph.
    /// </summary>
    /// <param name="serializationStream">
    /// Byte stream containing the serialized data.</param>
    /// <returns>A deserialized object graph.</returns>
    public object Deserialize(System.IO.Stream serializationStream)
    {
      return _formatter.Deserialize(serializationStream);
    }

    /// <summary>
    /// Converts an object graph into a byte stream.
    /// </summary>
    /// <param name="serializationStream">
    /// Stream that will contain the the serialized data.</param>
    /// <param name="graph">Object graph to be serialized.</param>
    public void Serialize(System.IO.Stream serializationStream, object graph)
    {
      _formatter.Serialize(serializationStream, graph);
    }

    #endregion

    /// <summary>
    /// Gets a reference to the underlying
    /// <see cref="BinaryFormatter"/>
    /// object.
    /// </summary>
    public BinaryFormatter Formatter
    {
      get
      {
        return _formatter;
      }
    }
  }
}
#endif